text/0000-rtsan-and-attributes.md

```diff
@@ -0,0 +1,273 @@
+`#[blocking]` defines a function as unfit for execution within a `#[nonblocking]` function.
+
+This example in the previous section shows that the interceptors written for the RealtimeSanitizer runtime are mostly shared across Rust and C/C++. Rust calls into libc `malloc` for basic allocation operations so it is automatically intercepted with no additional changes to the Rust version. A vast majority of the unsafe behavior that RTSan detects will be detected in this way.
+
+Users may also mark their own functions as unfit for a `nonblocking` context with the `#[blocking]` attribute, as seen below. This allows for detection of calls that do not result in a system call, but may be non-deterministically delayed.
+
+Both of these attributes
+
+```rust
+#[blocking]
+fn spin() {
+    loop {}
+}
+
+
+#[nonblocking]
+fn process() {
+    spin();
+}
+
+fn main() {
+    process();
+}
+> cargo run --package example --features rtsan
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.05s
+     Running `target/debug/example`
+==16364==ERROR: RealtimeSanitizer: blocking-call
+Call to blocking function `spin` in real-time context!
+    #0 0x0001021a6940 in rtsan::notify_blocking_call::ha111b6bcc1e1c566+0x24 (example:arm64+0x100002940)
+    #1 0x0001021a6864 in example::spin::h3f7c52e56f3fcffa main.rs:1
+    #2 0x0001021a6880 in example::process::hace23451997fd8f2 main.rs:9
+    #3 0x0001021a6840 in example::main::h700c0ae2eb60a977 main.rs:16
+    #4 0x0001021a670c in core::ops::function::FnOnce::call_once::hd7f394a2ba7ce532 function.rs:250
+    #5 0x0001021a6824 in std::sys::backtrace::__rust_begin_short_backtrace::h73bbd1f9991c56fb backtrace.rs:154
+    #6 0x0001021a67ec in std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::hf6a09edd6941dbc1 rt.rs:164
+    #7 0x0001021bebbc in std::rt::lang_start_internal::h9e88109c8deb8787+0x324 (example:arm64+0x10001abbc)
+    #8 0x0001021a67b8 in std::rt::lang_start::ha4c268826019738b rt.rs:163
+    #9 0x0001021a68b4 in main+0x20 (example:arm64+0x1000028b4)
+    #10 0x00018d8860dc  (<unknown module>)
+    #11 0x4c3d7ffffffffffc  (<unknown module>)
+
+SUMMARY: RealtimeSanitizer: blocking-call (example:arm64+0x100002940) in rtsan::notify_blocking_call::ha111b6bcc1e1c566+0x24
+fish: Job 1, 'cargo run --package example --f…' terminated by signal SIGABRT (Abort)
+
+```
+
+### The addition of the `rtsan_scoped_disabler!` macro
+
+It will be important to allow users to opt-out of rtsan detection for a specific scope. This may be useful if the end user thinks RTsan has a false positive, or it happens in third-party code they don't control.
+
+We propose addition of the `rtsan_scoped_disabler!` macro:
+```rust
+#[nonblocking]
+fn process() {
+  rtsan_scoped_disabler!({
+        let audio = vec![1.0; 256]; // report is suppressed
+  });
+}
+```
+
+When rtsan is not enabled, this will become a no-op, so it will be safe for users to leave in their code. 
+
+We are open to hearing which file this macro would belong in.
```

Comment:
Added - I wasn't sure where it should live, and the template recommends that we give specific implementation details when necessary. Our initial version had almost no specific implementation details